src/transformers/trainer.py

```diff
@@ -629,12 +811,15 @@ def train(self, model_path: Optional[str] = None, trial: Union["optuna.Trial", D
         # Model re-init
         if self.model_init is not None:
             # Seed must be set before instantiating the model when using model_init.
+            # XXX: move this and the same inside __init__ to call_model_init?
```

Comment:
No `model_init` needs to be called at each new `train` method, it's there to re-initialize the model at each new run for HP-search.
This comment should then be removed.